e2e/main.py

```diff
@@ -294,22 +326,89 @@ def call_bulk(bulks, *, batch_size):
+        # To successfully do this, we also need to either move the index price such that the
+        # cancellation is warranted or deposit extra collateral to the burrow. Here we do the latter.
+        for queued_leaf_ptr, leaf in auction_avl_leaves(
+            checker,
+            checker.storage["deployment_state"]["sealed"]["liquidation_auctions"][
+                "queued_slices"
+            ](),
+        ):
+            # Only retrieve the first queued slice ptr for efficiency
+            break
+        # Deposit a ton of tez to this burrow to ensure that it is no longer liquidatable
+        call_endpoint(
+            checker,
+            "deposit_tez",
+            leaf["leaf"]["value"]["contents"]["burrow"][1],
+            amount=100_000_000_000,
+        )
+        call_endpoint(checker, "cancel_liquidation_slice", queued_leaf_ptr)
+
+        # And we place a bid:
 
         ret = checker.metadata.currentLiquidationAuctionMinimumBid().storage_view()
         auction_id, minimum_bid = ret["contents"], ret["nat_1"]
         # The return value is supposed to be annotated as "auction_id" and "minimum_bid", I
         # do not know why we get these names. I think there is an underlying pytezos bug
         # that we should reproduce and create a bug upstream.
 
+        # Note the auction ptr for later operations
+        current_auctions_ptr = checker.storage["deployment_state"]["sealed"][
+            "liquidation_auctions"
+        ]["current_auction"]()["contents"]
         call_endpoint(
             checker, "liquidation_auction_place_bid", (auction_id, minimum_bid)
         )
 
+        # Once max(max_bid_interval_in_blocks, max_bid_interval_in_seconds) has elapsed, the liquidation auction we
+        # bid on should be complete. Here we go off of level since the patched version of Checker we expect to use
+        # sets max_bid_interval_in_blocks=1 and max_bid_interval_in_seconds=1.
+        # Sleeping a bit extra to add a bit of a buffer.
+        time.sleep(10)
+
+        call_endpoint(checker, "touch", None)
+        assert (
+            checker.storage["deployment_state"]["sealed"]["liquidation_auctions"][
+                "completed_auctions"
+            ]()
+            is not None
+        )
+
+        # Before we can claim our winning bid, all of the slices in the completed auction must be touched
+        auctioned_slices = []
+        for leaf_ptr, leaf in auction_avl_leaves(checker, current_auctions_ptr):
+            # Double check that we are only working with leaf ptrs
+            assert "leaf" in leaf
+            auctioned_slices.append(leaf_ptr)
+
+        # Note: batching here based on empirical estimates of how many slices we can
+        # touch in one go before hitting the gas limit.
+        for i in range(0, len(auctioned_slices), 10):
+            slices_to_cancel = auctioned_slices[i : i + 10]
+            call_endpoint(checker, "touch_liquidation_slices", slices_to_cancel)
```

Comment:
I wonder if we'd gain anything if we touched the slices a little out of order here, to cover more execution paths. It's certainly not necessary; just a thought :slightly_smiling_face:

Reply:
That's a good point. I think that ties in well with the goal of adding some profiling to these tests, so I noted it in a new issue here: https://github.com/tezos-checker/checker/issues/194.

Reply:
Nice! :+1: